lib/ansible/plugins/connection/network_cli.py

```diff
@@ -181,10 +181,10 @@ def _handle_prompt(self, resp, obj):
         """Matches the command prompt and responds"""
         if isinstance(obj, (binary_type, text_type)) or not isinstance(obj['prompt'], Sequence):
             obj['prompt'] = [obj['prompt']]
-        prompts = [re.compile(r, re.I) for r in obj['prompt']]
+        prompts = [re.compile(to_text(r, errors='surrogate_or_strict'), re.I) for r in obj['prompt']]
```

Comment:
This is likely wrong.  I believe that the changes that went in last week mean that obj['prompt'] is going to be a byte string.  resp is also a byte string.  So instead of converting both of them to text strings and risking either a traceback in converting (with surrogate_or_strict) or malforming the data (with surrogate_then_replace), it's better to compare them as byte strings.

I do see that the isinstance condition on line 182 allows for either a text or byte string, though.  If we want to continue to allow that then we'll have to make this